lib/fuse_lowlevel.c

```diff
@@ -1985,6 +1999,14 @@ static void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
 		return;
 	}
 
+	if (se->conn.max_write < bufsize - HEADER_SIZE) {
```

Comment:
This check is intended to cover the case where init() has *increased* `max_write`, right? If so, let's make that explicit by storing `old_max_write` and comparing with it after init() has run.

Reply:
max_write is at the maximum already. Just before calling init() max_write is adjusted to be the maximum that fits into the buffer. This check is in case it is not using the buffer completely. In that case we can safely remove the total size of the buffer.